build/Band_Representations.jl

```diff
@@ -0,0 +1,310 @@
+      
+     #  print("here3\n\n\n")    
+
+       #print(site_symmetry_index, "\n\n\n")
+
+        if site_symmetry_index >0
+            #println(tα′α′)
+            #print(exp(2*pi*1im*dot(transformed_k,tα′α′)))
+            χ_G_k_G=χ_G_k_G+exp(2im*π*dot(transformed_k, tα′α′))*tr(site_symmetry_irreps(sitegroup)[ρ].matrices[site_symmetry_index])
+            #print(transformed_k)
+            #print(dot(transformed_k, tα′α′))
+        end
+    end
+    return χ_G_k_G
+end
+
+
+function subduce_onto_littlegroups(k::String, kvec::KVec, ρ::Int64, sitegroup::SiteGroup{D}) where D
+    sgnum=sitegroup.num
+    num_lgirreps= size(get_lgirreps(sgnum, D)[k])[1]
+    
+    size_lgirreps=size(get_lgirreps(sgnum, D)[k][1].matrices)[1]
+
+    character_table=Matrix{Complex{Float64}}(undef, size_lgirreps, num_lgirreps )
+
+    for j in 1:num_lgirreps
+        for i in 1:size_lgirreps
+            character_table[i, j]=tr(get_lgirreps(sgnum, D)[k][j].matrices[i])
+        end
+    end
+    #return character_table
+    
+    #Now we obtain the vector that we require: 
+    XVector=Matrix{Complex{Float64}}(undef, (size_lgirreps, 1))
+
+    for i in 1:size_lgirreps
+        rep_index=group_representative(group(get_lgirreps(sgnum, D)[k][1])[i], spacegroup(sgnum, D)) 
+        iter=induced_band_representation(sitegroup, spacegroup(sgnum, D)[rep_index], ρ, kvec)
+       XVector[i, 1]= iter
+    end
+
+    #return round.(character_table\XVector)
+    #return character_table, XVector
+    return round.(character_table\XVector)
+
+end
+
+#StringtoKVec=Dict("Γ"=>KVec("0, 0"), "K"=>KVec("1/3, 1/3") , "M"=>KVec("1/2, 0"))
+
+function return_irvec(sgnum::Int64, ρ::Int64, sitegroup::SiteGroup{D}) where D
+    klab2kv = Dict(klab => kvec(lg) for (klab, lg) in get_littlegroups(sgnum, D))
+    irvec = Int[]
+    for (klab, kv) in klab2kv
+        isspecial(kv) || continue
+        kv_array   = subduce_onto_littlegroups(klab, kv, ρ, sitegroup)
+        append!(irvec, Int.(kv_array))
+    end
+    return irvec
+end
+
+function return_irvec2d(sgnum::Integer, D::Integer=2)
+    sg=spacegroup(sgnum, D);
+    wps=get_wycks(sgnum, D)
+    A=Vector{Vector{Int}}(undef, 0)
+
+    sitegroups=SiteGroup.(Ref(sg), wps)
+
+    maximal_wyckoffs=Array{WyckPos{2}, 1}(undef, 0)
+
+    for maximal_sitegroup in findmaximal(sitegroups)
+        push!(maximal_wyckoffs, maximal_sitegroup.wp)
+    end
+
+    #for wp in wps
+    for wp in maximal_wyckoffs
+       # print(wp)
+        #isspecial(qvec(wp)) || continue 
+        #isspecial(wp) || continue 
+
+        sitegroup=SiteGroup(sg, wp)
+        ρ_max=size(site_symmetry_irreps(sitegroup))[1]
+        #print(ρ_max)
+        for ρ in 1:ρ_max
+            push!(A, return_irvec(sgnum, ρ, sitegroup))
+        end
+    end
+    return hcat(A...)
+end
+
+
+function get_maximal_lgirreps(sgnum::Integer, D::Integer)
+    maximal_lgirreps=Dict{String, Array{LGIrrep{D}, 1}}()
```

Comment:
```suggestion
    maximal_lgirreps = Dict{String, Vector{LGIrrep{D}}}()
```
Same thing, just more readable.